actionview/lib/action_view/template.rb

```diff
@@ -328,7 +325,8 @@ def handle_render_error(view, e) #:nodoc:
       end
 
       def locals_code #:nodoc:
-        locals = @locals.delete_if { |key| TEMPLATE_RESERVED_METHOD_NAMES.include?(key.downcase) }
+        locals = @locals.to_set - Module::DELEGATION_RESERVED_METHOD_NAMES
```

Comment:
iirc Array `-` already compares by hash for speed. The set conversion shouldn't be necessary.